nomt/src/lib.rs

```diff
@@ -291,14 +310,49 @@ impl<T: HashAlgorithm> Nomt<T> {
             session_cnt: self.session_cnt.clone(),
             metrics: self.metrics.clone(),
             rollback_delta,
-        }
+        })
+    }
+
+    /// Update the merkle tree and commit the changes to an in-memory [`Overlay`].
+    ///
+    /// The actuals are a list of key paths and the corresponding read/write operations. The list
+    /// must be sorted by the key paths in ascending order. The key paths must be unique.
+    pub fn update(
+        &self,
+        _session: Session,
+        _actuals: Vec<(KeyPath, KeyReadWrite)>,
+    ) -> anyhow::Result<Overlay> {
+        unimplemented!()
+    }
+
+    /// Update the merkle tree and commit the changes to an in-memory [`Overlay`] and create a
+    /// proof of all updated values.
+    ///
+    /// The actuals are a list of key paths and the corresponding read/write operations. The list
+    /// must be sorted by the key paths in ascending order. The key paths must be unique.
+    pub fn update_and_prove(
+        &self,
+        _session: Session,
+        _actuals: Vec<(KeyPath, KeyReadWrite)>,
+    ) -> anyhow::Result<(Overlay, Witness, WitnessedOperations)> {
+        unimplemented!()
     }
 
-    /// Commit the transaction and returns the new root.
+    /// Commit the changes from this overlay to the underlying database.
+    ///
+    /// This assumes no sessions are active and will panic otherwise.
+    ///
+    /// The caller should take care to commit overlays in order and not to commit conflicting
+    /// overlays. Otherwise, the database will be left in an inconsistent state.
```

Comment:
I don't love this, and I'm considering enforcing overlays to be committed in sequence. It seems super error prone to just corrupt the DB silently.